packages/framework/tsup.config.ts

```diff
@@ -1,18 +1,22 @@
-import { defineConfig } from 'tsup';
-import { type SupportedFrameworkName } from './src';
+import { type Options, defineConfig } from 'tsup';
+import { version } from './package.json';
+import { type SupportedFrameworkName } from './src/internal';
 
 const frameworks: SupportedFrameworkName[] = ['h3', 'express', 'next', 'nuxt', 'sveltekit', 'remix', 'lambda', 'nest'];
 
 const baseConfig: Options = {
   entry: ['src/index.ts', 'src/internal/index.ts', ...frameworks.map((framework) => `src/servers/${framework}.ts`)],
   sourcemap: false,
   clean: true,
-  treeshake: true,
   dts: true,
   minify: true,
```

Comment:
Libraries shouldn't be minified. The host application bundler should handle that. 

However, setting `bundle: false` and `minify: false` in `tsup` config, breaks Nest.js loading @novu/framework. 

The hypothesis is that building through ESBuild alone is not enough. Minify triggers a rollup processing that, for some reason, satisfies Nest.js CommonJS requirements. 

We will get back to this in a separate PR, as this one is already big enough.

Reply:
Thanks for the detailed explanation. Sounds like a good plan 👍